test/setup/iam.tf

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  log_export_required_roles = [
+    # Needed for the Pubsub submodule to create a service account for the
+    # subscription it creates
+    "roles/iam.serviceAccountAdmin",
+
+    # Needed for the cloud storage submodule to create/delete a bucket
+    "roles/storage.admin",
+
+    # Needed for the pubsub submodule to create/delete a pubsub topic
+    "roles/pubsub.admin",
+
+    # Needed for the bigquery submodule to create/delete a bigquery dataset
+    "roles/bigquery.dataOwner",
+
+    # Needed for the root module to activate APIs
+    "roles/serviceusage.serviceUsageAdmin",
+
+    # Needed for the Pubsub submodule to assign roles/bigquery.dataEditor to
+    # the service account it creates
+    "roles/resourcemanager.projectIamAdmin",
+
+    # Required to create log sinks from the project level
+    "roles/logging.configWriter",
+  ]
+
+  log_export_billing_account_roles = [
+    # Required to associate billing accounts to new projects
+    "roles/billing.user",
+  ]
+
+  log_export_organization_roles = [
+    # Required to create log sinks from the organization level on down
+    "roles/logging.configWriter",
+
+    # Required to associate billing accounts to new projects
+    "roles/billing.projectManager",
+  ]
+
+  log_export_folder_roles = [
+    # Required to spin up a project within the log_export folder
+    "roles/resourcemanager.projectCreator",
```

Comment:
Again, where are we creating a new project besides in the setup phase?

Reply:
Yes, this is meant for the setup phase so the required project for running tests can be created.

Reply:
Right, so that's not required. Please try to understand how this actually executes instead of putting in unnecessary permissions.

`test/setup` is applied using the Cloud Build service account. So permissions needed *only* to run the setup stage do NOT need to be added here.

Reply:
@morgante, sorry for confusion. I just double-checked that with this [force-push](https://github.com/terraform-google-modules/terraform-google-log-export/compare/21c7079f7f4585d8304ce8988f0ec8ecffc2fdf8..0a1a2127b56cdaf4cd2f36e4c5d69ea37a12ed78). The permission is required for tests them-self for the computed values test suite and can't take it out.
It's used [here](https://github.com/ivankorn/terraform-google-log-export/blob/29/test/fixtures/computed_values/main.tf#L23-L28)
![image](https://user-images.githubusercontent.com/50832464/65894932-46eb7d00-e3b3-11e9-99c6-a536936dbe7e.png)